### PR TITLE
Add Time to the class of values that can be used in a cell.  Avoid error when value is passed in options

### PR DESCRIPTION
**Pull request by petergoldstein**
This is a combination of #103 and #119 with tests for both cases.  Thanks to @schlick and @tetsuya-ogawa for the original PRs.

@pointlessone Since the original PRs overlap, and tests were only included in one of them, I think this is a good combination PR.  Missing tests were added.

It also makes it easy to add future types to be stringified and, should it be desired, to change the behavior to stringify anything but `Prawn::Table`, `Prawn::Table::Cell`,  or Array that responds to a `to_s`

@gettalong any comments are welcome as well.

### CODE REVIEW

File: lib/prawn/table/cell.rb [hunk 1]
```diff
@@ -191,6 +187,15 @@ def self.make(pdf, content, options={})
         end
       end
 
+      def self.stringify_content?(content)
+        return true if content.nil?
+        return true if content.kind_of?(Numeric)
+        return true if content.kind_of?(Date)
+        return true if content.kind_of?(Time)
```
Comment (gettalong) [inline lib/prawn/table/cell.rb:194]: Why do you use separate if statements instead of ||-ing them together?

**Review by gettalong — COMMENTED** [review]
You are introducing a slight behaviour change since the `#to_s` conversion is now also done on content provide via options Hash. 

However, I think that this is okay.

File: lib/prawn/table/cell.rb (hunk 1 above)
Comment (petergoldstein) [reply lib/prawn/table/cell.rb:194]: Readability / simpler conditions.  If there's a desire to concatenate I've got no huge objection.

**Comment by petergoldstein** [comment]
@gettalong Yeah, I was aware of the slight behavior change but I don't think it's an issue because:

1. We were already executing that change in the case where `options[:content]` was `nil` - `options[:content]` was being set to the empty string.
2. There doesn't appear to be a downstream case where the content being a Date/Time/Numeric was functional.  All of those look like they should get routed to the Text cell case.

File: lib/prawn/table/cell.rb (hunk 1 above)
Comment (gettalong) [reply lib/prawn/table/cell.rb:194]: It's fine as it is. Just wanted to know the reason.

File: spec/cell_spec.rb [hunk 2]
```diff
@@ -36,20 +36,19 @@ def cell(options={})
...
       end
     end
 
-    it "should allow inline styling with a hash argument" do
-      # used for table([[{:text => "...", :font_style => :bold, ...}, ...]])
-      c = Prawn::Table::Cell.make(@pdf,
-                                  {:content => 'hello', :font_style => :bold})
-      expect(c).to be_a_kind_of Prawn::Table::Cell::Text
-      expect(c.content).to eq "hello"
-      expect(c.font.name).to eq 'Helvetica-Bold'
```
Comment (pointlessone) [inline spec/cell_spec.rb:52]: Why did you remove this spec?

File: spec/cell_spec.rb [hunk 3]
```diff
@@ -36,20 +36,19 @@ def cell(options={})
...
+    it "should convert nil, Numeric, and Date values to strings when value is extracted from options" do
+      [nil, 123, 123.45, Date.today, Time.new].each do |value|
+        c = Prawn::Table::Cell.make(@pdf, {}, { :content => value })
+        expect(c).to be_a_kind_of Prawn::Table::Cell::Text
+        expect(c.content).to eq value.to_s
+      end
```
Comment (pointlessone) [inline spec/cell_spec.rb:51]: This looks a lot like the previous spec. Does it cover anything not covered by the other one?

**Review by pointlessone — COMMENTED** [review]
This seem to override #103. Do you still want #103 to be merged?

File: spec/cell_spec.rb (hunk 2 above)
Comment (petergoldstein) [reply spec/cell_spec.rb:52]: That was unintentional.  I'll put it back.

File: spec/cell_spec.rb (hunk 3 above)
Comment (petergoldstein) [reply spec/cell_spec.rb:51]: This spec tests an alternate path to the previous spec.  It's the whole point of #119.

The previous spec tests normalization of the passed in content argument, if it's not a Hash.  But there's an alternate path for getting the content value that was not being tested for normalization.  And the previous code wasn't normalizing it.

It content is a hash then the code:

1. Checks to see if this is supposed to be an image and, if so, returns that
2. Otherwise, merges the content hash into options
3. And then uses the :content value from that merged hash

In the existing code the value retrieved from (3) is not stringified when it should be.  Hence the error.  This spec is necessary to test this path.

**Comment by petergoldstein** [comment]
@pointlessone This does override #103.  If we merge this PR (which is my preference), we can close #103 .  Given that @schlick - the original author of #103 - gave this a thumbs up, I assume that he's good with that approach as well.

**Review by pointlessone — APPROVED** [review]
